lua/typescript-tools/init.lua

```diff
@@ -15,8 +17,17 @@ function M.setup(config)
 
   plugin_config.load_settings(settings)
 
+  -- Extract LSP config options (on_attach, capabilities, etc.)
+  -- Filter out plugin-specific options like 'settings'
+  local lsp_config = {}
+  for key, value in pairs(config) do
+    if key ~= "settings" then
+      lsp_config[key] = value
+    end
+  end
```

Comment:
this broke my previous change to store lsp config under config.config... what do we actually prefer? because this is actually supported in config.config